langchain4j-core/src/main/java/dev/langchain4j/store/embedding/filter/comparison/IsEqualTo.java

```diff
@@ -48,6 +50,10 @@ public boolean test(Object object) {
             return compareAsBigDecimals(actualValue, comparisonValue) == 0;
         }
 
+        if (comparisonValue instanceof UUID && actualValue instanceof String) {
```

Comment:
I'm not fan, but if we do not find a way to serialized UUID as UUID and not string. I think it could be the hack.